src/bin/sol-fbp-generator/main.c

```diff
@@ -233,8 +232,7 @@ read_meta_type_file(const struct sol_flow_metatype_context *ctx,
     }
 
     contents = sol_file_reader_get_all(reader);
-    *buf = contents.data;
-    *size = contents.len;
+    *buf = SOL_BUFFER_INIT_CONST((char *)contents.data, contents.len);
```

Comment:
Ok. To save you some time in the understanding: the purpose of `sol_blob` is to reference count read only memory and use a custom function to release it. It may have a parent to keep "views" of data, in this case the child keeps a reference to parent. Releasing the last reference of the child will unref its parent.

With that in mind we created blobs with various free functions (or no_data_free/no_free to not free the data -- to be used in views -- and not free the blob at all, like when the blob itself is part of something else). One of them is a blob to a file reader, that keeps the file opened and the memory mmap()ed, so the file is no allocated and can be evicted from RAM even if no swap is present.

Then if the `read_file` is changed to return a blob, once that blob is unreferenced, it would munmap()/close() right when it's not used anymore. There is no need to keep the list of opened files to be closed once the program is closed.

Functions that are processing "text" shouldn't take a blob, but a str_slice. Unless they need to keep the memory alive after they return.